test/marvelus/function_test.zls

```diff
@@ -0,0 +1,123 @@
+  let z = x *. y in
+  z
+
+(* Example 4: Division Function (fail) *)
+(*
+   The divide function takes two non-zero integer values, `x` and `y`,
+   and returns a float value `z` such that `z = x / y`.
+   This function should pass the refinement verification.
+    # (=>(and (= r z) (= z (/ x y))  (>= y 0.0) (= x 0.0)) (or(<= r x) (>= r x))
+   Constraints:
+   - `x` and `y` are non-zero integers.
+   - The return value `z` is the division of `x` by `y`.
+*)
+let divide (x : {v:float | v > 0 }) (y : {v:float | v > 0 }) : {r: float | r <= x } =
+  let z = x /. y in
+  z
+
+(* Example 5: Division Function with negative values (fail) *)
+(*
+   The divide function takes two negative integer values, `x` and `y`,
+   and returns a float value `z` such that `z = x / y`.
+   This function should fail the refinement verification.
+    # (=>(and (= r z) (= z (/ x y))  (not (= y 0.0)) (not (= x 0.0))) (or (<= r x) (>= r x))
+   Constraints:
+   - `x` and `y` can be negative integers.
+   - The return value `z` is the division of `x` by `y`.
+*)
+let divide_pos_neg (x : {v:float | v > 0 || v < 0 }) (y : {v:float | v > 0  || v < 0}) : {r: float | r <= x} =
+  let z = x /. y in
+  z
+
+(* Entry Point *)
+(*
+   The node function serves as the entry point for the test.
+   It calls the `add` , `sub` and `multiply` functions with sample arguments,
+   prints the results, and returns unit ().
+*)
+
+(* Example 6: Function with some intermediate variables (fail) *)
+(*
+   The some_compute function takes two positive float values, `x` and `y`,
+   and returns a float value `z` such that `z = d - 2.0` and d = c / 3.0 and c = b * 2.0 and b = a + 10.0 and a = x + y.
+   This function should pass the refinement verification.
+    # (=>(and (= v z) (= z (- d 2.0)) (= d (/ c 3.0)) (= c (/* b 2.0)) (= b (+ a 10.0)) (= a (+ x y)) (> y 0.0) (> x 0.0)) (and(>= z x) (>= z y))
+   Constraints:
+   - `x` and `y` must be positive floats.
+   - The return value `z` should satisfy the constraints `x <= z` and `y <= z`.
+*)
+
+let some_compute (x : {v:float|v>0.0} ) (y : {v:float|v>0.0} ) : { v:float | x <= v && y <= v } =
+  let a = x +. y in
+  let b = a +. 10.0 in
+  let c = b *. 2.0 in
+  let d = c /. 3.0 in
+  let z = d -. 2.0 in
+  z
+
+(* Example 7: Function with multiple operations (pass) *)
+(*
+   The multi_opertation_compute function takes two positive float values, `x` and `y`,
+   and returns a float value `z` such that `z = 2.0 * x + sqrt(x * x - 4.0 * y) / 2.0`.
+   This function should pass the refinement verification.
+    # (=>(and (= v z) (= z (+ (/* 2.0 x) (/ (sqrt (- (/* x x) (/* 4.0 y))) 2.0))) (> y 0.0) (> x 0.0)) (and (>= z 0.0))
+   Constraints:
+   - `x` and `y` must be positive floats.
+   - The return value `z` should satisfy the constraint `z >= 0.0`.
+*)
+let multi_opertation_compute (x : {v:float|v>0.0} ) (y : {v:float|v>0.0} ) : { v:float | v >= 0.0 } =
+  let z = (2.0 *. x) +. (sqrt (x *. x -. 4.0 *. y)) /. 2.0 in
```

Comment:
I don't think `sqrt` is implemented yet in the verifier, which is causing `42` to appear in some constraints. This is a separate issue that also needs to be addressed.